orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala

```diff
@@ -150,6 +160,7 @@ object Ec2Resource {
     subnetId: String,
     instanceType: String,
     instanceProfile: String,
+    tags: Option[Seq[AwsTag]],
```

Comment:
Seems this can just be a sequence of tuples to simplify things?

Reply:
@schowsf would play pass validation if tags are defined as optional tuples with missing value?

Reply:
without the case class for AwsTag; using Option[Seq[Tuple2[String, String]]] for spec.tag 

// happy path
`"tags":[["Environment","qa"],["mce-compute",""],["host_role","eef-compute"]]`,

// missing value for "Environment" key
`"tags":[["Environment"],["mce-compute",""],["host_role","eef-compute"]]`,

```
[error] 2022-06-07 14:17:38,287 c.s.m.o.s.a.ResourceMgr$ - Invalid resource spec
com.salesforce.mce.orchard.system.util.InvalidJsonException: field: /tags(0), error: List(JsonValidationError(List(Expected array of 2 elements),List()))
```

Reply:
@realstraw  preference on the following to model tags? 

1. case class -> more descriptive error
2. tuple -> simpler payload

Reply:
what about `Seq[(String, Option[String])]`, I believe aws tag values are optional?

Reply:
values can be empty string, but cannot be null.

Reply:
ok, I'm good with either object or tuple then. One more question, `tags: Option[Seq[AwsTag]]`, does it have to be `Option[Seq]`? can it be simply `Seq`?

Reply:
I think Option will allow for the payload to not have tags at all, or `"tags": null`

the more normal payloads would have tags like 
```"tags":[{"key":"Environment"},{"key":"mce-conmpute","value":""},{"key":"host_role","value":"eef-compute"}]```

I slightly prefer to the case class, so the json validation error is a little more descriptive.  Plus gives me a place to write a few words as comments on the case class `AwsTag`.

Reply:
got it, make sense